esp32c3-hal/examples/aes_dma.rs

```diff
@@ -0,0 +1,154 @@
+//! Encrypt/Decrypt a message using AES
+
+#![no_std]
+#![no_main]
+use aes::{
+    cipher::{generic_array::GenericArray, BlockDecrypt, BlockEncrypt, KeyInit},
+    Aes128 as Aes128SW,
+};
+use esp32c3_hal::{
+    aes::{
+        dma::{CipherMode, WithDmaAes},
+        Aes,
+        Mode,
+    },
+    clock::ClockControl,
+    dma::DmaPriority,
+    gdma::Gdma,
+    peripherals::Peripherals,
+    prelude::*,
+    systimer::SystemTimer,
+};
+use esp_backtrace as _;
+use esp_println::println;
+
+#[entry]
+fn main() -> ! {
+    let peripherals = Peripherals::take();
+    let system = peripherals.SYSTEM.split();
+    let _clocks = ClockControl::boot_defaults(system.clock_control).freeze();
+
+    let dma = Gdma::new(peripherals.DMA);
+    let dma_channel = dma.channel0;
+
+    let mut descriptors = [0u32; 8 * 3];
+    let mut rx_descriptors = [0u32; 8 * 3];
+
+    let aes = Aes::new(peripherals.AES).with_dma(dma_channel.configure(
+        false,
+        &mut descriptors,
+        &mut rx_descriptors,
+        DmaPriority::Priority0,
+    ));
+
+    let keytext = buffer1();
+    let plaintext = buffer2();
+    plaintext.copy_from_slice(&[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16]);
+    keytext.copy_from_slice(&[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16]);
+
+    // create an array with aes128 key size
+    let mut keybuf = [0_u8; 16];
+    keybuf[..keytext.len()].copy_from_slice(keytext);
+
+    // create an array with aes block size
+    let mut block_buf = [0_u8; 16];
+    block_buf[..plaintext.len()].copy_from_slice(plaintext);
+
+    let hw_encrypted = buffer3();
+    let pre_hw_encrypt = SystemTimer::now();
+    let transfer = aes
+        .dma_transfer(
```

Comment:
not sure if the name is really good - while it does a DMA transfer it's probably not the interesting part for the public API